src/tests/system/tests/test_cache.py

```diff
@@ -412,3 +416,62 @@ def test_cache__both_ldap_user_email_and_extra_attribute_email_are_stored(client
     assert user_dict["description"] == ["gecos1"], "attribute 'description' was not correct"
     assert user_dict["mail"] == ["user1@example.test"], "attribute 'mail' was not correct"
     assert user_dict["email"] == ["user1@example.test"], "attribute 'email' was not correct"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_cache__sssd_cache_nowait_percentage(client: Client, provider: GenericProvider):
+    """
+    :title: Test SSSD netgroup caching with 'entry_cache_nowait_percentage'
+    :setup:
+        1. A netgroup is created and member is added to the netgroup
+        3. SSSD is configured with entry_cache_nowait_percentage
+        4. The domain "test" is updated with `entry_cache_timeout` set to 30 seconds
+        5. SSSD is restarted with the `clean=True` option to apply the new configuration
+    :steps:
+        1. Record the initial response time for netgroup query
+        2. Introduce a network delay of 50 seconds
+        3. Execute another query for netgroup
+        4. Record the response time and verify it is faster than the initial query
+        5. Check the SSSD NSS logs for the presence of the log entry:
+            "Performing midpoint cache update of [netgrp_nowait@test]"
+    :expectedresults:
+        1. The query should return the netgroup with the correct name
+        2. Network delayed for 50 seconds
+        3. Query is executed
+        4. The response time for subsequent queries should be faster than the initial query
+        6. The expected log entries are present in the logs
+    :customerscenario: True
+    """
+    if not isinstance(provider, (LDAP, Samba, AD)):
+        pytest.skip("IPA does not support domain in netgroups")
+
+    log_nss = "/var/log/sssd/sssd_nss.log"
+    netgroup_group = provider.netgroup("netgrp_nowait").add()
+    netgroup_group.add_member(host="host1", user="kau10", domain="example.com")
+    client.sssd.nss.update(
+        filter_groups="root", filter_users="root", debug_level="9", entry_cache_nowait_percentage="50"
+    )
+    client.sssd.dom("test").update(entry_cache_timeout="30")
```

Comment:
You should lower this value to the minimum possible to avoid getting stuck in this test for too long

Reply:
Is 30 the minimum value?

Reply:
yes, for this test we must

Reply:
Can you state the reasons? I'd like to understand why this is happening

Reply:
Also sssd.conf says.

If the domain's entry_cache_timeout is set to 30s and entry_cache_nowait_percentage is set to 50 (percent), entries that come in after 15 seconds past the last cache update will be returned immediately, but
 Valid values for this option are 0-99 and represent a percentage of the entry_cache_timeout for each domain. For performance reasons, this percentage will never reduce the nowait timeout to less than 10 seconds.